app/master/cluster_master.py

```diff
@@ -213,6 +213,7 @@ def handle_request_for_new_build(self, build_params):
         if build_request.is_valid():
             build = Build(build_request)
             self._all_builds_by_id[build.build_id()] = build
+            build.generate_project_type()
```

Comment:
I had made this same comment on Joey's PR when he had an initialize() method in addition to his constructor. He pointed out that it's bad practice to do any significant work in the constructor other than variable assignment for a few reasons.

The biggest one is that it's not expected. When you instantiate an object, you don't expect the constructor to throw an exception or to take a lot of computing resources to do--you expect it to always return you an instance quickly.

The other reason was for testability--as it is harder to test the constructor of objects.